subsys/ipc/ipc_service/backends/ipc_icmsg.c

```diff
@@ -26,7 +26,7 @@ struct backend_data_t {
 	struct icmsg_buf *rx_ib;
 
 	/* Backend ops for an endpoint. */
-	const struct ipc_service_cb *ops;
+	const struct ipc_ept_cfg *cfg;
```

Comment:
to be honest I would prefer if priv is separate

as @rakons noticed we cannot make ops const static, placed in rom and reused for all instances because configuration structure holds priv parameter for each instance.

anyway this is a wider problem that maybe be fixed for all backends.

Reply:
As discussed offline - this is general problem in many places in zephyr API style. Not only here.

Reply:
@rakons i think its worthwhile to report enhancement issue so that we don't forget about it in the future.

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/44187